### DISCUSSION TITLE
Dropwizard and OpenAPI/Swagger

### DISCUSSION THREAD

**Discussion by rit001** [dropwizard/dropwizard#7531] [Q&A]
I'm new to the whole Dropwizard stack, but I have gone as far as being able to deploy a 4.0.x test environment, so I can say that I understand the basic environment.

My next step is to look at generating Rest API docs from the code, but have currently hit something of a brick wall. Both Dropwizard and Swagger have changed a lot over the years, but all the examples of how to integrate them are very dated and so nonfunctional.

Does anyone know of an up-to-date write-up of the work required to combine these 2 projects?

Thanks in advance

**Comment by josephlbarnett** [comment]
https://github.com/trib3/leakycauldron/blob/main/server/src/main/kotlin/com/trib3/server/swagger/SwaggerInitializer.kt may be helpful (kotlin code but sets up swagger on the admin servlet with the contents of the main jax-rs application)?  

It's been a while since I've looked at it but last I did it worked with dw 3.0.x and swagger 2.2.15.  dw 4.0.x/jakarta namespace probably makes things more complicated but i think there are swagger-*-jakarta modules that you can use?

**Comment by Athou** [comment]
What I'm doing for [CommaFeed](https://github.com/Athou/commafeed) is
- use `io.swagger.core.v3:swagger-annotations` to [annotate the REST endpoints ](https://github.com/Athou/commafeed/blob/master/commafeed-server/src/main/java/com/commafeed/frontend/resource/OpenAPI.java)
- use `io.swagger.core.v3:swagger-maven-plugin-jakarta` to [generate an openapi.json file](https://github.com/Athou/commafeed/blob/c2e0c81f7eebca2a45cf50b422fed9e36cc1ed8f/commafeed-server/pom.xml#L146-L169) at _build_ time and put it on the classpath so it's available at runtime
- expose the [openapi.json](https://www.commafeed.com/openapi.json) file with a dropwizard `AssetsBundle`
- use [Redoc](https://www.commafeed.com/#/api/) to generate the documentation at runtime

Hope this helps :)

**Reply by badeavali** [reply]
@Athou 

I've tried following your steps, annotated the classes, methods etc but I have a huge problem at step 2: using that plugin to generate the openapi.json file.

The reason, is that I am using gradle(kotlin) instead of maven.

Is there any advice you'd have in this direction? (For context, I'm in a dropwizard environment)
